src/meta/src/model/mod.rs

```diff
@@ -670,6 +671,264 @@ impl<'a, K: Ord, V: PartialEq + Transactional<TXN>, TXN> ValTransaction
     }
 }
 
+pub enum BTreeMapTransactionWrapper<'a, K: Ord, V> {
```

Comment:
We can implement `Deref` and `DerefMut` for the wrapper instead of the explicit `tree` and `tree_mut`, and we don't have to manually implement the series of btreemap methods.

Code can be like
```
impl Deref BTreeMapTransactionWrapper<'a, K, V> {
    type Target = BTreeMap<K, V>;

    fn deref(&self) -> &Self::Target {
        match self {
            BTreeMapTransactionWrapper::V1(v) => v.tree_ref,
            BTreeMapTransactionWrapper::V2(v) => v.tree_ref,
        }
    }
}
```

Reply:
> we don't have to manually implement the series of btreemap methods.

I think this doesn't work, because those methods need to merge `staging` and `tree_ref`.